webui/public/js/bootstrap-table-sort.js

```diff
@@ -0,0 +1,38 @@
+/**
+ *
+ * bareos-webui - Bareos Web-Frontend
+ *
+ * @link      https://github.com/bareos/bareos for the canonical source repository
+ * @copyright Copyright (c) 2013-2023 Bareos GmbH & Co. KG (http://www.bareos.org/)
+ * @license   GNU Affero General Public License (http://www.gnu.org/licenses/)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+function sortSemanticVersion (a, b) {
+    var i, diff;
+    var regExStrip0 = /(\.0+)+$/;
+    var segmentsA = a.replace(regExStrip0, '').split('.');
+    var segmentsB = b.replace(regExStrip0, '').split('.');
+    var l = Math.min(segmentsA.length, segmentsB.length);
+
+    for (i = 0; i < l; i++) {
+        diff = parseInt(segmentsA[i], 10) - parseInt(segmentsB[i], 10);
+        if (diff) {
+            return diff;
+        }
+    }
+    return segmentsA.length - segmentsB.length;
+}
```

Comment:
```suggestion
function compareSegments(a, b)
{
    // 1a < 1 < 1.0
    // 1abc < 1xyz < 1 < 2abc < 2
    // 1~pre1.abc < 1~pre2.abc < 1~pre10.abc
    if (a == b) {
    	return 0;
    }

    if (isNaN(a) && isNaN(b)) {
      // a: str, b: str
    	if (a <= b) {
      	return -1;
      } else {
      	return 1;
      }
    } else if (isNaN(a)) {
      // a: str, b: number
    	return -1;
    } else if (isNaN(b)) {
      // a: number, b: str
    	return 1;
    } else {
      // a: number, b: number
      return parseInt(a) - parseInt(b);
    }
}

function getSegments(str)
{
	return str.match(/[\d]+|[^\d]+|\./g).filter(seg => seg != '.');
}

function sortSemanticVersion(a, b) {
	var segmentsA = getSegments(a);
  var segmentsB = getSegments(b);
  var len = Math.min(segmentsA.length, segmentsB.length);

  for (var i = 0; i < len; i++) {
  	var result = compareSegments(segmentsA[i], segmentsB[i]);
    if (result != 0) {
    	return result;
    }
  }
  return segmentsA.length - segmentsB.length;
}
```
I tested your sort function and found a few glitches, especially with `~pre` versions. Writing this function took me more time than expected, but I thing it covers all our version strings.